### PR TITLE
#8294: Set moment library language during app setup

### LINKED ISSUE #8294: Make use of moment library localization support
**Is your feature request related to a problem? Please describe.**
Superset uses an external library _moment_, for date and time management. As the library supports localization, Superset could make use of it, and set its language when needed.   

**Describe the solution you'd like**
I suggest using moment.locale method to change its language globally (as described in https://momentjs.com/docs/#/i18n/changing-locale/), when the app is set up (which is happening also on language change).

### PR DESCRIPTION
**Pull request by oskarhinc**
### CATEGORY
Choose one
- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Make use of moment.locale method to change its language globally (as described in https://momentjs.com/docs/#/i18n/changing-locale/), when the app is set up (which is happening also on language change).

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API
- [x] Fixes: #8294

### CODE REVIEW

File: superset/assets/src/setup/setupApp.js [hunk 1]
```diff
@@ -20,6 +20,16 @@
 import $ from 'jquery';
 import { SupersetClient } from '@superset-ui/connection';
 import getClientErrorObject from '../utils/getClientErrorObject';
+import moment from 'moment';
+import localStorage from 'local-storage';
```
Comment (kristw) [inline superset/assets/src/setup/setupApp.js]: Why is this needed instead of using the native `localStorage`?

File: superset/assets/src/setup/setupApp.js [hunk 2]
```diff
@@ -20,6 +20,16 @@
...
+
+const DEFAULT_D3_LANGUAGE = 'en';
+
+function applyLocales() {
+  const currentLocale = localStorage.get('locale');
```
Comment (kristw) [inline superset/assets/src/setup/setupApp.js]: Where is the code that set the `locale`?

File: superset/assets/src/setup/setupApp.js (hunk 1 above)
Comment (oskarhinc) [reply superset/assets/src/setup/setupApp.js]: You're right, it's not needed in this case - I'm just used to using it.

File: superset/assets/src/setup/setupApp.js (hunk 2 above)
Comment (oskarhinc) [reply superset/assets/src/setup/setupApp.js]: Nice catch, I haven't included it - I'll add it in next commit there: https://github.com/apache/incubator-superset/blob/1dd4d7a587a55525ff14e1e051d268adccbc15a1/superset/assets/src/preamble.js#L31

**Comment by oskarhinc** [comment]
@kristw I've corrected mentioned issues.
Regarding Travis checks - are they always reliable? I'm wondering as in previous builds there were fails in other places than in the latest one (due to request timeout as well), but it doesn't seem to be related with my changes in code.

**Comment by villebro** [comment]
@oskarhinc some tests aren't always reliable; I restarted CI, check back in a while to see if it fixes the problem.

**Comment by oskarhinc** [comment]
@villebro Thanks a lot!

File: superset/assets/src/preamble.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ if (typeof window !== 'undefined') {
   if (bootstrapData.common && bootstrapData.common.language_pack) {
     const languagePack = bootstrapData.common.language_pack;
     configure({ languagePack });
+    localStorage.setItem('locale', bootstrapData.common.locale);
```
Comment (kristw) [inline superset/assets/src/preamble.js]: Does it produce the same result if you call `moment.locale()` here without using `localStorage` at all?

Comment (oskarhinc) [reply superset/assets/src/preamble.js]: You're right - I've removed the usage of `localStorage` now

**Review by kristw — APPROVED** [review]
